pkg/eval/builtin_fn_str_test.go

```diff
@@ -50,7 +51,35 @@ func TestWcswidth(t *testing.T) {
 
 func TestEawk(t *testing.T) {
 	Test(t,
+		That(`
+			echo "1\n2\n3\n4" | eawk [@a]{
```

Comment:
1. These `echo` commands are worth rewriting to `put line1 line2 | to-lines`; with a lot of "\n" the strings are a bit hard to read.
2. Add comments on which behavior of `eawk` is tested by each test case.